### PR TITLE
Add Kotlin solution for factorial

### PR DESCRIPTION
**Pull request by IrgendNJulian**
First time using GitHub. Hope I did everything right.

### CODE REVIEW

**Comment by mre** [comment]
Hey @IrgendNJulian,
thanks for your submissions. I don't have much time at the moment to maintain this project as well as I wanted, so I prefer to get smaller pull requests for single-file changes if possible.

The code looks good overall.
I'd merge it, but I would like to revert the change on `problems/can-scramble/can-scramble.kt` before. The existing solution is actually quite nice.

```
length == other.length && chars().sum() == other.chars().sum()
```

The idea is that you don't have to sort the characters to know if two strings can be scrambled. It's enough to compare the lengths and then to sum up the characters to create some kind of "hash" for the string, which is unique for every permutation of the string.
In total, this is an `O(n)` operation whereas sorting would be `O(n*log n)`.

So, could you undo the change on that file?
The rest of the code looks good. 😃

**Comment by mre** [comment]
Ping. Any update on this?

**Comment by IrgendNJulian** [comment]
Ouh. Sorry for not responding to your first comment. 
In the old version of the Kotlin CanScramble file "ac".canScramble("bb") returned true since 'a' + 'c' = 97 + 99 = 196 = 98 + 98 = 'b' + 'b' but should have returned false.
But i just looked it up in the repo and it has already been fixed by now.
So, undo will be done.

File: problems/factorial/factorial.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+fun factorial(n: Int): Int =
+    when (n) {
+        0, 1 -> 1
+        else -> n * factorial(n - 1)
+    }
```
Comment (Fleshgrinder) [inline problems/factorial/factorial.kt]: We now have a more comprehensive solution for this in Kotlin, see #131.

**Comment by Fleshgrinder** [comment]
I’m closing this because we have solutions for both problems by now. @IrgendNJulian please open another PR that takes the current solutions into account if you think that the existing solutions can be improved. :blush:

**Pull request closed without merging**